tangram/src/lib.rs

```diff
@@ -0,0 +1,559 @@
+	#[cfg_attr(feature = "std", derive(Debug))]
+	pub struct TangramRealmMetadata<Hash,BlockNumber> {
+		realm_id: Hash,
+		name: Vec<u8>,
+		cid: Vec<u8>, // ipfs Content ID
+		created: BlockNumber,
+		mutated: BlockNumber,
+	}	
+
+
+
+
+	/// TangramClass
+	#[derive(Encode, Decode, Default, PartialEq, Eq, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(Debug))]
+	pub struct TangramClass<Hash, /*Balance*/> {
+		id: Hash,	// unique class hash
+		realm: RealmIndex,// auth realm
+		index: u64,	// class index
+		max: u64,	// max number of items in class
+		// mint: Balance, // mint cost
+		// burn: Balance, // burn cost
+		strategy: u64, // cid to strategy //?
+	}
+
+	/// TangramClassMetadata
+	#[derive(Encode, Decode, Default, PartialEq, Eq, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(Debug))]
+	pub struct TangramClassMetadata<Hash, BlockNumber> {
+		class_id: Hash,
+		name: Vec<u8>,
+		cid: Vec<u8>, //content id in ipfs
+		created: BlockNumber,
+		mutated: BlockNumber,
+		f: Vec<u8> //?
+	}
+
+	/// Tangram Immutable + Unique
+	#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+	pub struct TangramItem<Hash, MomentOf> {
+		dna: Hash,
+		dob: MomentOf,
+	}
+
+	/// Tangram Mutable
+	#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Default, RuntimeDebug, TypeInfo)]
+	pub struct TangramItemMetadata {
+		realm: RealmIndex,
+		class: ClassIndex,
+		name: Vec<u8>,
+		cid: Vec<u8>,
+	}
+
+
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// Storage
+
+	// Realms
+	#[pallet::storage]
+	#[pallet::getter(fn next_realm_index)]
+	/// pub NextRealmIndex get(fn next_realm_index): RealmIndex = 0;
+	pub type NextRealmIndex<T: Config> = StorageValue<_, u64, ValueQuery>; // <- todo check
+
+	#[pallet::storage]
+	#[pallet::getter(fn realm_by_hash)]
+	/// pub Realm get(fn realm_by_hash): map hasher(blake2_128_concat) T::Hash => TangramRealm<T::Hash>;
+	pub(super) type Realm<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, TangramRealm<T::Hash>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn realm_by_index)]
+	///pub RealmByIndex get(fn realm_by_index): map hasher(blake2_128_concat) RealmIndex => T::Hash;
+	pub(super) type RealmByIndex<T: Config> = StorageMap<_, Blake2_128Concat, RealmIndex, T::Hash, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn owner_for_realm)]
+	///pub OwnerRealm get(fn owner_for_realm): map hasher(blake2_128_concat) RealmIndex => T::Hash;
+	pub(super) type OwnerRealm<T: Config> = StorageMap<_, Blake2_128Concat, RealmIndex, T::Hash, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn realms_for_owner)]
+	/// pub RealmsForOwner get(fn realms_for_owner): map hasher(blake2_128_concat) T::Hash => Vec<T::Hash>;
+	pub(super) type RealmsForOwner<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, Vec<T::Hash>, ValueQuery>;
+
+
+	#[pallet::storage]
+	#[pallet::getter(fn realms_for_owner_count)]
```

Comment:
We've decided to get rid of the getters until it's not needed from the outside and use full syntax inside of the pallet instead, ex.: `RealmsForOwnerCount::<T>::get()`